### PR TITLE
Add RGB encoder

### PR DESCRIPTION
**Pull request by saturday06**

### CODE REVIEW

**Comment by saturday06** [comment]
rebased

File: webp/rgb_image.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+func (p *RGBImage) RGBAAt(x, y int) color.RGBA {
+	if !(image.Point{x, y}.In(p.Rect)) {
+		return color.RGBA{}
+	}
+	i := (y-p.Rect.Min.Y)*p.Stride + (x-p.Rect.Min.X)*3
+	return color.RGBA{p.Pix[i+0], p.Pix[i+1], p.Pix[i+2], 0xFF}
+}
+
+// ColorModel is RGB color model instance
+var ColorModel = color.ModelFunc(rgbModel)
```
Comment (harukasan) [inline webp/rgb_image.go]: `ColorModel` is ambiguous. `RGBModel` is better.

Comment (saturday06) [inline webp/rgb_image.go]: 🆗 
Renamed.

File: webp/rgb_image.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+// RGBAAt returns the color of the pixel at (x, y) as RGBA.
+func (p *RGBImage) RGBAAt(x, y int) color.RGBA {
+	if !(image.Point{x, y}.In(p.Rect)) {
+		return color.RGBA{}
+	}
+	i := (y-p.Rect.Min.Y)*p.Stride + (x-p.Rect.Min.X)*3
+	return color.RGBA{p.Pix[i+0], p.Pix[i+1], p.Pix[i+2], 0xFF}
+}
+
+// ColorModel is RGB color model instance
```
Comment (harukasan) [inline webp/rgb_image.go]: s/ColorModel/RGBModel/

File: webp/rgb_image.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// Package rgb provides RGB image which implements image.Image interface.
+package webp
+
+import (
+	"image"
+	"image/color"
+)
+
+// Image represent image data which has RGB colors.
```
Comment (harukasan) [inline webp/rgb_image.go]: Comment should starts with RGBImage

File: webp/rgb_image.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	r |= r << 8
+	g = uint32(c.G)
+	g |= g << 8
+	b = uint32(c.B)
+	b |= b << 8
+	a = uint32(0xFFFF)
+	return
+}
+
+// Make sure Image implements image.Image.
```
Comment (harukasan) [inline webp/rgb_image.go]: s/Image/RGBImage/

Comment (saturday06) [inline webp/rgb_image.go]: Fixed

File: webp/rgb_image.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
+// Package rgb provides RGB image which implements image.Image interface.
```
Comment (harukasan) [inline webp/rgb_image.go]: Remove package comment.

Comment (saturday06) [inline webp/rgb_image.go]: oops

**Pull request merged**